### PR TITLE
Perceptual metrics

### PR DESCRIPTION
**Pull request by danielbarco**

### CODE REVIEW

File: src/chuchichaestli/metrics/__init__.py
Comment (bil-y) [inline src/chuchichaestli/metrics/__init__.py:1]: This file needs a header comment (just the license text) to make the linter happy.

Comment (bil-y) [reply src/chuchichaestli/metrics/__init__.py:1]: The same is true for the other new files. If you copy a file from somehwere else, keep their header and just add ours above or below, stating that "the file below is copied from XYZ and modifications are licensed under our license".

File: src/chuchichaestli/metrics/backbones/clip.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        if pretrained_weights is None:
+            pretrained_weights = ARCH_WEIGHT_DEFAULTS[arch]
+
+        self.model = open_clip.create_model(arch, pretrained_weights)
+        self.clean_resize = clean_resize
+        self.depth = depth
+
+    def transform(self, image):
+        mean = (0.48145466, 0.4578275, 0.40821073)
+        std = (0.26862954, 0.26130258, 0.27577711)
```
Comment (bil-y) [inline src/chuchichaestli/metrics/backbones/clip.py:39]: Where do these numbers come from? Can you comment above them and make them configurable? I.e., use these values as default but make it possible to set different ones.

File: src/chuchichaestli/metrics/backbones/swav.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+try:
+    from torchvision.models.utils import load_state_dict_from_url
+except ImportError:
+    from torch.utils.model_zoo import load_url as load_state_dict_from_url
+from torchvision import transforms
+
+from chuchichaestli.metrics.backbones.backbone_encoder import Encoder
+
+SWAV_WEIGHTS_URL = (
+    "https://www.dropbox.com/s/ghjrgktg0lyjorn/swav_800ep_pretrain.pth.tar?dl=1"
```
Comment (bil-y) [inline src/chuchichaestli/metrics/backbones/swav.py:28]: Is this your Dropbox or someone else's? If it isn't yours, could you upload it to SwitchDrive or somewhere where we have more control over it?

File: tests/test_lpips.py
Comment (phdenzel) [inline tests/test_lpips.py:1]: Tests are all skipped, when not on Nvidia GPU... so setting device to "cpu" is redundant. Will add a more flexible skip condition...

File: tests/test_frechet_distance.py
Comment (phdenzel) [inline tests/test_frechet_distance.py:1]: Tests are all skipped, when not on Nvidia GPU... so setting device to "cpu" is redundant. Will add a more flexible skip condition...

File: src/chuchichaestli/metrics/lpips.py
Comment (phdenzel) [inline src/chuchichaestli/metrics/lpips.py:1]: Lots of bloat code... should we reduce the unnecessary bits?

File: src/chuchichaestli/metrics/frechet_distance.py
Comment (phdenzel) [inline src/chuchichaestli/metrics/frechet_distance.py:1]: See comment for `lpips.py`

File: src/chuchichaestli/metrics/backbones/inception.py [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+import torchvision.transforms.functional as TF
+
+
+from chuchichaestli.metrics.backbones.backbone_encoder import Encoder
+
+from chuchichaestli.metrics.backbones.resizer import pil_resize
+
+# Inception weights ported to Pytorch from
+# http://download.tensorflow.org/models/image/imagenet/inception-2015-12-05.tgz
+FID_WEIGHTS_URL = "https://github.com/mseitzer/pytorch-fid/releases/download/fid_weights/pt_inception-2015-12-05-6726825d.pth"  # noqa: E501
```
Comment (phdenzel) [inline src/chuchichaestli/metrics/backbones/inception.py:24]: See @bil-y comment on the pretrained weights URLs: https://github.com/CAIIVS/chuchichaestli/pull/87/files#r1888447482

File: src/chuchichaestli/metrics/backbones/clip.py (hunk 1 above)
Comment (phdenzel) [reply src/chuchichaestli/metrics/backbones/clip.py:39]: Those are the infamous ImageNet mean and variances...

**Comment by danielbarco** [comment]
I have submitted a clean up issue here: https://github.com/CAIIVS/chuchichaestli/issues/91

**Comment by phdenzel** [comment]
I'll resolve all comments and finish the review... we'll continue on #91

**Review by phdenzel — APPROVED** [review]
Code is functional, which is the main goal right now... rest will be dealt with in #91

**Pull request merged**